### ISSUE TITLE
Why use the intermediate estimated x_0|t as the distribution matching input， but not the final image of generator?

### ISSUE THREAD

**Issue by vv12kant** [tianweiy/DMD2#45]
As far as I understand，the input of ditribution mathing should be the final output of generator, but under the circumstance of multi-step generator, we use the intermediate estimated x_0|t as the input，why?

**Comment by tianweiy** [comment]
you are right. However, using the final output requires us to back propagate through the time (or step) which will adds more gpu memory requirement. It should be possible to do this some system optimization but we didn't get chance to try because of this concern

**Comment by vv12kant** [comment]
> you are right. However, using the final output requires us to back propagate through the time (or step) which will adds more gpu memory consumption. It should be possible to do this with some system optimization but we didn't get chance to try because of this concern
Thanks. I think I understand something: the multi-step generator is like an LCM model. The difference is that LCM needs to align the final output x_0 of all steps on the ODE, while DMD needs to match the distribution at all steps on the ODE.

**Comment by vv12kant** [comment]
Thanks. I think I understand something: the multi-step generator is like an LCM model. The difference is that LCM needs to align the final output x_0 of all steps on the ODE, while DMD needs to match the distribution at all steps on the ODE.
